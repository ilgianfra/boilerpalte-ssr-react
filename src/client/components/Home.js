import React from 'react';
import { Query } from 'react-apollo';
import styled from 'styled-components';

import { EXAMPLE_QUERY } from '../../graphql/querys';

const H3 = styled.h3`
background: #944a62;
padding: 2px;
border-radius: 3px;
border: none;
width: fit-content
color: white;`;

class Home extends React.Component {

  render() {
    return (
      <Query query={EXAMPLE_QUERY}>
        {({ loading, error, data }) => {
          if (loading) {
            return <h1>Loading</h1>;
          }
          if (error) {
            return <h1>{`Error ${error.message}`}</h1>;
          }
          return data.rates.map((element) => {
            return (
              <H3 key={element.currency}>{`The ${element.name}'s short name is ${element.currency}`}</H3>
            );
          });
        }}
      </Query>
    );
  }
}

// export default {
//   component: Home
// };
export default Home;
