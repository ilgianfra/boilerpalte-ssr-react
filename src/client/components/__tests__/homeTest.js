import React from 'react';

// import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from '../Home';

describe('Home', () => {
  configure({ adapter: new Adapter() });
  const testProps = {
    name: 'Home'
  };
  it('renders correctly', () => {
    const wrapper = shallow(
      <Home
        {...testProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
