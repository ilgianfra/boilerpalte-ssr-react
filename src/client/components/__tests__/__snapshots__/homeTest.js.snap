// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    name="Home"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello from Home component!",
    },
    "ref": null,
    "rendered": "Hello from Home component!",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello from Home component!",
      },
      "ref": null,
      "rendered": "Hello from Home component!",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
